Upstream-Status: Backport

* Backported from the ToT: https://crrev.com/c/1296495

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From ea08f8b130a355d19ebc4fc62f2937ff4466ce6e Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Tue, 23 Oct 2018 16:50:41 +0000
Subject: [PATCH 03/17] [ozone/common] Fix IsValidBufferFormat.

Currently IsValidBufferFormat translates fourcc format
to a gfx::BufferFormat using GetBufferFormatFromFourCCFormat method.

But if GetBufferFormatFromFourCCFormat does not know about a
particular fourcc format, it just crashes. Thus, it makes
more sense to check for a supported buffer format using
fourcc drm formats instead.

Change-Id: If904b2f8b31a6cefdcc41940d69fe3d2112c246b
Reviewed-on: https://chromium-review.googlesource.com/c/1296495
Reviewed-by: Michael Spang <spang@chromium.org>
Commit-Queue: Maksim Sisov <msisov@igalia.com>
Cr-Commit-Position: refs/heads/master@{#601985}
---
 ui/ozone/common/linux/drm_util_linux.cc | 26 ++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/ui/ozone/common/linux/drm_util_linux.cc b/ui/ozone/common/linux/drm_util_linux.cc
index b9a26887dffd..48648b703d1a 100644
--- a/ui/ozone/common/linux/drm_util_linux.cc
+++ b/ui/ozone/common/linux/drm_util_linux.cc
@@ -75,19 +75,19 @@ gfx::BufferFormat GetBufferFormatFromFourCCFormat(int format) {
 }
 
 bool IsValidBufferFormat(uint32_t current_format) {
-  switch (GetBufferFormatFromFourCCFormat(current_format)) {
-    case gfx::BufferFormat::R_8:
-    case gfx::BufferFormat::RG_88:
-    case gfx::BufferFormat::RGBA_8888:
-    case gfx::BufferFormat::RGBX_8888:
-    case gfx::BufferFormat::BGRA_8888:
-    case gfx::BufferFormat::BGRX_8888:
-    case gfx::BufferFormat::BGRX_1010102:
-    case gfx::BufferFormat::RGBX_1010102:
-    case gfx::BufferFormat::BGR_565:
-    case gfx::BufferFormat::UYVY_422:
-    case gfx::BufferFormat::YUV_420_BIPLANAR:
-    case gfx::BufferFormat::YVU_420:
+  switch (current_format) {
+    case DRM_FORMAT_R8:
+    case DRM_FORMAT_GR88:
+    case DRM_FORMAT_ABGR8888:
+    case DRM_FORMAT_XBGR8888:
+    case DRM_FORMAT_ARGB8888:
+    case DRM_FORMAT_XRGB8888:
+    case DRM_FORMAT_XRGB2101010:
+    case DRM_FORMAT_XBGR2101010:
+    case DRM_FORMAT_RGB565:
+    case DRM_FORMAT_UYVY:
+    case DRM_FORMAT_NV12:
+    case DRM_FORMAT_YVU420:
       return true;
     default:
       return false;
-- 
2.17.1

